src/sentry/api/serializers/models/repository.py

```diff
@@ -25,13 +25,19 @@ def serialize(self, obj, attrs, user):
                 'id': 'unknown',
                 'name': 'Unknown Provider',
             }
+
+        if obj.status == ObjectStatus.PENDING_DELETION:
+            repo_name = obj.config.get('pending_deletion_name', obj.name)
```

Comment:
@markstory I saved the name in the config for repository so that we can show that to the user instead if it is pending deletion. That was we don't run into the field size limitiations that @mattrobenolt mentioned but also we can show something to the user that makes sense.

Reply:
I like this approach :+1: It strikes a good balance on usability and doesn't require an additional OrganizationOption record.